File: Lib/Telerik/Examples/FileExplorer/CustomButton.vb
Imports System.Text
Imports Telerik.WinControls.UI
Imports Telerik.WinControls.Primitives

Namespace FileExplorer
	Friend Class CustomButton
		Inherits RadButtonElement
		Protected Overrides Sub OnMouseEnter(ByVal e As EventArgs)
            Me.BorderElement.Visibility = Telerik.WinControls.ElementVisibility.Visible
            Me.ButtonFillElement.ResetValue(FillPrimitive.BackColorProperty, Telerik.WinControls.ValueResetFlags.Local)
            Me.ButtonFillElement.ResetValue(FillPrimitive.BackColor2Property, Telerik.WinControls.ValueResetFlags.Local)
            Me.ButtonFillElement.ResetValue(FillPrimitive.BackColor3Property, Telerik.WinControls.ValueResetFlags.Local)
            Me.ButtonFillElement.ResetValue(FillPrimitive.BackColor4Property, Telerik.WinControls.ValueResetFlags.Local)
			MyBase.OnMouseEnter(e)
			Console.WriteLine("reset")
		End Sub

		Protected Overrides Sub OnMouseLeave(ByVal e As EventArgs)
            Me.ButtonFillElement.BackColor = Color.Transparent
            Me.ButtonFillElement.BackColor2 = Color.Transparent
            Me.ButtonFillElement.BackColor3 = Color.Transparent
            Me.ButtonFillElement.BackColor4 = Color.Transparent
            Me.ButtonFillElement.Visibility = Telerik.WinControls.ElementVisibility.Collapsed
			MyBase.OnMouseLeave(e)
			Console.WriteLine("transparent")
		End Sub

		Protected Overrides ReadOnly Property ThemeEffectiveType() As Type
			Get
				Return GetType(RadButtonElement)
			End Get
		End Property
	End Class
End Namespace

File: Lib/Telerik/Examples/FileExplorer/CustomButton.cs
using System;
using System.Collections.Generic;
using System.Text;
using Telerik.WinControls.UI;
using System.Drawing;
using Telerik.WinControls.Primitives;

namespace FileExplorer
{
    class CustomButton : RadButtonElement
    {
        protected override void OnMouseEnter(EventArgs e)
        {
            this.BorderElement.Visibility = Telerik.WinControls.ElementVisibility.Visible;
            this.ButtonFillElement.ResetValue(FillPrimitive.BackColorProperty, Telerik.WinControls.ValueResetFlags.Local);
            this.ButtonFillElement.ResetValue(FillPrimitive.BackColor2Property, Telerik.WinControls.ValueResetFlags.Local);
            this.ButtonFillElement.ResetValue(FillPrimitive.BackColor3Property, Telerik.WinControls.ValueResetFlags.Local);
            this.ButtonFillElement.ResetValue(FillPrimitive.BackColor4Property, Telerik.WinControls.ValueResetFlags.Local);
            base.OnMouseEnter(e);
            Console.WriteLine("reset");
        }

        protected override void OnMouseLeave(EventArgs e)
        {
            this.ButtonFillElement.BackColor = Color.Transparent;
            this.ButtonFillElement.BackColor2 = Color.Transparent;
            this.ButtonFillElement.BackColor3 = Color.Transparent;
            this.ButtonFillElement.BackColor4 = Color.Transparent;
            this.BorderElement.Visibility = Telerik.WinControls.ElementVisibility.Collapsed;
            base.OnMouseLeave(e);
            Console.WriteLine("transparent");
        }

        protected override Type ThemeEffectiveType
        {
            get
            {
                return typeof(RadButtonElement);
            }
        }
    }
}
